src/PhotoGallery/AbstractPhotoTrigger.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <QObject>
+
+/**
+ * \brief A "photo" operation in progress
+ *
+ * Represents a photo being taken. This object exists while this is in
+ * progress.
+ */
+class AbstractPhotoTriggerOperation : public QObject {
+    Q_OBJECT
+public:
+    ~AbstractPhotoTriggerOperation() override;
+
+    /**
+     * \brief Whether this operation has finished.
+     */
+    virtual bool finished() const = 0;
```

Comment:
The Coding Style for comments is to use ```///``` style comments not ```/* .. */```. Also don't add comments for things which are obvious by there method naming. Like here. Commenting that this represents "finished" doesn't add any value and just takes up room on the header file.

https://github.com/mavlink/qgroundcontrol/blob/master/CodingStyle.h
https://github.com/mavlink/qgroundcontrol/blob/master/CodingStyle.cc

Reply:
Reformatted for coding style.

I am not clear about the intended difference between /// and // -- I take it one is exported docstrings, the other isn't? In that case the way this is set now is as intended.

I did *not* follow the "_" for protected/private methods. This makes a mess as methods a) change visibility during development and b) via "using" in subclasses. The redudancy means too many pointless changes to me.

Reply:
> I am not clear about the intended difference between /// and // -- I take it one is exported docstrings, the other isn't? In that case the way this is set now is as intended.

QGC coding style is to use ```//``` commenting versus ```/* ... */```. For doxygen the coding style is to use ```///```.

> I did _not_ follow the "_" for protected/private methods. This makes a mess as methods a) change visibility during development and b) via "using" in subclasses. The redudancy means too many pointless changes to me.

It is important to know whether a method is public or private from a readability standpoint to understand the scoping of the method without needing to look at the header. Same thing for "_foo" class member variable naming. Also I would consider moving a method from private to public requiring more work a good thing. Since it make you look at the code in a deeper way to make sure it's ok to move this to public.

Reply:
> > I am not clear about the intended difference between /// and // -- I take it one is exported docstrings, the other isn't? In that case the way this is set now is as intended.
> 
> QGC coding style is to use `//` commenting versus `/* ... */`. For doxygen the coding style is to use `///`.
> 
> > I did _not_ follow the "_" for protected/private methods. This makes a mess as methods a) change visibility during development and b) via "using" in subclasses. The redudancy means too many pointless changes to me.
> 
> It is important to know whether a method is public or private from a readability standpoint to understand the scoping of the method without needing to look at the header. Same thing for "_foo" class member variable naming. Also I would consider moving a method from private to public requiring more work a good thing. Since it make you look at the code in a deeper way to make sure it's ok to move this to public.

The class member variable naming is sensible, the visibility is not.

FTR, none of the 'major' coding styles is doing that (boost / standard C++, Google, Qt, ...) and some of them have published extensive rationale explaining the reasoning for choices. I am not clear what makes QGC "exceptional" or where some of the C++ experts behind those are missing crucial insight from QGC development. One particular thing for certain is that the convention is logically at odds with the language as any private/protected inheritance and/or using directives changing visibility render the name invalid. 

For the class 'user' it does not matter as the visibility is enforced by compiler without naming convention. For class 'implementor' it still does not matter as there is no difference in calling either, and the implementor needs to know about the detailed semantics of each called function anyways which cannot be conveyed in a single character.

There is a gradual line of moving between "member functions", "static member functions" and "free functions" where any such naming conventions is contributing useless red tape. (Free functions cannot be legally named with an underscore). There is also a gradual process in defining the member exposure of implementation(!) classes when working on unit test coverage, again adding nothing but red tape.

That said, I am going to change implementation according to style, but it is not going to change my opinion that it is entirel pointless.

Reply:
I'll take all of that into account in considering a change to the coding style.